postgre-client/src/asset_index_client.rs

```diff
@@ -244,92 +269,182 @@ pub(crate) fn split_assets_into_components(asset_indexes: &[AssetIndex]) -> Asse
-        let mut transaction = self.start_transaction().await?;
         let table_names = TableNames {
             metadata_table: "tasks".to_string(),
             assets_table: "assets_v3".to_string(),
             creators_table: "asset_creators_v3".to_string(),
             authorities_table: "assets_authorities".to_string(),
-            fungible_tokens_table: "fungible_tokens".to_string(),
         };
-        self.upsert_batched(&mut transaction, table_names, updated_components)
-            .await?;
-        self.commit_transaction(transaction).await
+        let operation_start_time = chrono::Utc::now();
+        let mut transaction = self.start_transaction().await?;
+
+        // Perform transactional operations
+        let result = self
+            .upsert_batched_nft(&mut transaction, table_names, updated_components)
+            .await;
+
+        match result {
+            Ok(_) => {
+                // Commit the transaction and record the time taken
+                self.commit_transaction(transaction).await?;
+                self.metrics.observe_request(
+                    SQL_COMPONENT,
+                    TRANSACTION_ACTION,
+                    "transaction_total",
+                    operation_start_time,
+                );
+                Ok(())
+            }
+            Err(e) => {
+                // Transaction will be rolled back automatically when `transaction` goes out of scope
+                self.metrics.observe_request(
+                    SQL_COMPONENT,
+                    TRANSACTION_ACTION,
+                    "transaction_failed_total",
+                    operation_start_time,
+                );
+                Err(e)
+            }
+        }
+    }
+
+    async fn update_fungible_asset_indexes_batch(
+        &self,
+        fungible_asset_indexes: &[FungibleAssetIndex],
+    ) -> Result<(), IndexDbError> {
+        let operation_start_time = chrono::Utc::now();
+        let mut transaction = self.start_transaction().await?;
+        let table_name = FungibleTokenTable::new("fungible_tokens");
+
+        // Perform transactional operations
+        let result = self
+            .upsert_batched_fungible(
+                &mut transaction,
+                table_name,
+                split_into_fungible_tokens(fungible_asset_indexes),
+            )
+            .await;
+
+        match result {
+            Ok(_) => {
+                // Commit the transaction and record the time taken
+                self.commit_transaction(transaction).await?;
+                self.metrics.observe_request(
+                    SQL_COMPONENT,
+                    TRANSACTION_ACTION,
+                    "transaction_total",
+                    operation_start_time,
+                );
+                Ok(())
+            }
+            Err(e) => {
+                // Transaction will be rolled back automatically when `transaction` goes out of scope
```

Comment:
well... until it will not 
![image](https://github.com/user-attachments/assets/40e880e3-ab69-41d0-b311-d94688ef123b)
![image](https://github.com/user-attachments/assets/dd3ef7cd-e447-4502-96de-b02ce7086d51)
![image](https://github.com/user-attachments/assets/1bbf8fc3-de91-4d07-bb14-9fc759efb646)

Reply:
Some of the transactions on the screens were started by a process that has already been stopped/killed for many hours. And those were blocking the other transactions that were waiting for the lock held by the first txs. All this comes somewhere from async with those txs and probably bugs/issues either in the sqlx or our code leaking some of the arcs somewhere